### PR TITLE
Adding KRM and TF alternatives for BigQuery snippets

### PR DESCRIPTION
**Pull request by AlexBulankou**
This is the first PR to add KRM and TF alternatives for BigQuery snippets. It can be used for subsequent PRs as example.

1. See go/dm-snippets-alternatives for objective and background
2. Adding readme to show how to launch each of the tools
3. Adding tf and krm directories with configs
4. Adding test_alternatives.sh intended to run manually
5. Adding create_projects and delete_projects scripts that can be used to simplify testing

### CODE REVIEW

File: google/resource-snippets/bigquery-v2/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+gcloud container clusters get-credentials [CLUSTER_ID] --zone=[ZONE]
+cd alternatives/krm
+kpt cfg set . deployment d1
+kubectl apply -f bigquery.yaml
+```
+
+## Testing
+
+```bash
+sh test_alternatives.sh
```
Comment (andrei-scripniciuc) [inline google/resource-snippets/bigquery-v2/README.md:32]: could you mark this file as executable? I think this should work for github `git update-index --chmod=+x google/resource-snippets/bigquery-v2/test_alternatives.sh`

File: tools/alt-testing/create-projects.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+# Expecting
+# TF_PROJECT_ID=
+# KRM_PROJECT_ID=
+# DM_PROJECT_ID=
+
+BILLING_ACCOUNT=[BILLING_ACCOUNT]
+FOLDER_ID=[FOLDER_ID]
+CLUSTER_ID=[CLUSTER_ID]
+ZONE=[ZONE]
```
Comment (andrei-scripniciuc) [inline tools/alt-testing/create-projects.sh:11]: looks like all of these variables are required, is the intention that the ones that are commented out are set as env vars while the remaining must be set in the script? can we be more consistent?

File: tools/alt-testing/create-projects.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
+set -e
```
Comment (andrei-scripniciuc) [inline tools/alt-testing/create-projects.sh:1]: this file should be marked as executable as well, suggest running `git update-index --chmod=+x tools/alt-testing/create-projects.sh`

File: tools/alt-testing/delete-projects.sh [hunk 4]
```diff
@@ -0,0 +1,3 @@
+gcloud projects delete -q ${DM_PROJECT_ID}
```
Comment (andrei-scripniciuc) [inline tools/alt-testing/delete-projects.sh:1]: this file should be marked as executable as well, suggest running git update-index --chmod=+x `tools/alt-testing/delete-projects.sh`

File: tools/alt-testing/create-projects.sh (hunk 2 above)
Comment (andrei-scripniciuc) [reply tools/alt-testing/create-projects.sh:11]: I see that the project variables are referenced in `delete-project.sh` as well, it makes sense for those to be global in that case. perhaps just add a comment confirming that it's intentional that some variables must be defined in the script while others should be global

File: google/resource-snippets/bigquery-v2/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Bigquery Snippets
+
```
Comment (andrei-scripniciuc) [inline google/resource-snippets/bigquery-v2/README.md:2]: could you add a short setup section mentioning which tools are required? from a quick scan I saw

* `gcloud`
* `gsutil`
* `kubectl`
* `terraform`
* `kpt`

File: tools/alt-testing/README.md [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# This directory contains the scripts that can be used for testing Terraform and config-connector alternatives
+
```
Comment (andrei-scripniciuc) [inline tools/alt-testing/README.md:2]: could you add a short setup section mentioning which tools are required? from a quick scan I saw

* `gcloud`
* `gsutil`
* `kubectl`
* `terraform`
* `kpt`

File: google/resource-snippets/bigquery-v2/README.md (hunk 1 above)
Comment (AlexBulankou) [reply google/resource-snippets/bigquery-v2/README.md:32]: done

File: tools/alt-testing/create-projects.sh (hunk 2 above)
Comment (AlexBulankou) [reply tools/alt-testing/create-projects.sh:11]: I added the comment in the file to clarify - this script can be included as source and the variables above are coming from the parent `test_alternatives` script, while the 3 other variables should be defined here.

File: tools/alt-testing/create-projects.sh (hunk 3 above)
Comment (AlexBulankou) [reply tools/alt-testing/create-projects.sh:1]: done

File: tools/alt-testing/delete-projects.sh (hunk 4 above)
Comment (AlexBulankou) [reply tools/alt-testing/delete-projects.sh:1]: done

**Review by andrei-scripniciuc — APPROVED** [review]

**Pull request merged**